src/components/eip4626/user/MintAssetToken.tsx

```diff
@@ -0,0 +1,171 @@
+import { useReadHtsTokenTokenAddress } from "@/hooks/eip4626/useReadHtsTokenTokenAddress";
+import { useWriteHtsTokenAssociate } from "@/hooks/eip4626/mutations/useWriteHtsTokenAssociate";
+import {
+  Alert,
+  AlertDescription,
+  AlertIcon,
+  AlertTitle,
+  Button,
+  Text,
+} from "@chakra-ui/react";
+import { EvmAddress, VaultMintTokenProps } from "@/types/types";
+import {
+  DEFAULT_TOKEN_MINT_AMOUNT,
+  useWriteHtsTokenMint,
+} from "@/hooks/eip4626/mutations/useWriteHtsTokenMint";
+import { useReadTokenDecimals } from "@/hooks/eip4626/useReadTokenDecimals";
+import { formatBalance, formatNumberToBigint } from "@/services/util/helpers";
+import { useReadBalanceOf } from "@/hooks/useReadBalanceOf";
+import { useAccountTokens } from "@/hooks/useAccountTokens";
+import { useEffect, useState } from "react";
+import { AccountId } from "@hashgraph/sdk";
+import { useQueryClient } from "@tanstack/react-query";
+import { QueryKeys } from "@/hooks/types";
+
+export function MintAssetToken({ vaultAssetSelected }: VaultMintTokenProps) {
+  const queryClient = useQueryClient();
+
+  //@TODO add readTokenName to show vault asset token names
+  const { data: deployedHtsTokensAddress } =
+    useReadHtsTokenTokenAddress(vaultAssetSelected);
+
+  const { data: vaultAssetSelectedDecimals } = useReadTokenDecimals(
+    deployedHtsTokensAddress,
+  );
+
+  const {
+    data: associateResult,
+    mutateAsync: associate,
+    error: associateError,
+    isPending: isAssociatePending,
+  } = useWriteHtsTokenAssociate();
+
+  const {
+    data: mintResult,
+    mutateAsync: mint,
+    error: mintError,
+    isPending: isMintPending,
+  } = useWriteHtsTokenMint();
+
+  const { data: tokenBalance, error: tokenBalanceError } = useReadBalanceOf(
+    deployedHtsTokensAddress as EvmAddress,
+  );
+
+  const {
+    data: accountTokens,
+    isFetching,
+    hasNextPage,
+    fetchNextPage,
+  } = useAccountTokens();
+
+  useEffect(() => {
+    if (hasNextPage && !isFetching) {
+      fetchNextPage();
+    }
+  }, [accountTokens, fetchNextPage, hasNextPage, isFetching]);
```

Comment:
Why `accountTokens` added here?
I mean even we need it nothing to deal with any checks.

Reply:
nice catch